src/averageAges.js

```diff
@@ -15,13 +15,35 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  if (century === undefined) {
+    const arrOfMens = people.filter(item => item['sex'] === 'm');
+    const sumOfMensAge = arrOfMens.reduce((startValue, person) => {
+      return startValue + (person['died'] - person['born']);
+    }, 0);
+    const mensAverageAge = sumOfMensAge / arrOfMens.length;
+
+    return mensAverageAge;
+  }
+
+  const arrOfMens2 = people.filter(item => item['sex'] === 'm');
+  const diedInCentury = arrOfMens2.filter(
+    (person) => {
+      return Math.ceil(person['died'] / 100) === century;
+    },
+  );
+  const sumOfCentury = diedInCentury.reduce(
+    (startValue, person) => startValue + (person['died'] - person['born']), 0
+  );
+  const averageOfCentury = sumOfCentury / diedInCentury.length;
+
+  return averageOfCentury;
```

Comment:
there are a lot of repeated logic
you can prepare initial array (filtered by century or not) and remain the same for both at the end
DTY (do not repeat yourself)